Source/VCGeneration/Split.cs

```diff
@@ -736,6 +737,173 @@ public Counterexample ToCounterexample(ProverContext context)
         throw new cce.UnreachableException();
       }
 
+      private static void PrintSet<T> (HashSet<T> s) {
+        foreach(T i in s)
+          Console.WriteLine(i);
+      }
+
+      private static Dictionary<Block, Block> ImmediateDominator(List<Block> blocks)
```

Comment:
I did not understand why `immediateDominatorMap` returns a map from Block to a list of Blocks. Probably because it computes all dominators? So I would need to get the *last* dominator from it, topologically speaking.

Also, the algorithm in `Graph.cs` is for general graphs and I have a DAG here; DAGs make things simpler and more efficient (only need one pass through the graph).

That said, I don't have a good sense of how important the difference is. So if you think it won't matter, I can use the one in `Graph.cs`.

Reply:
Please move your fast algorithm to Graph.cs as a public method. Document its difference with the other one.  Then we can proceed to landing this PR.